src/mixed.js

```diff
@@ -115,12 +131,21 @@ const proto = (SchemaType.prototype = {
 
     var next = prependDeep(schema.clone(), this);
 
-    // new undefined default is overriden by old non-undefined one, revert
+    // new undefined default is overridden by old non-undefined one, revert
     if (has(schema, '_default')) next._default = schema._default;
 
     next.tests = this.tests;
     next._exclusive = this._exclusive;
 
+    // manually merge the blacklist/whitelist (the other `schema` takes
+    // precedence in case of conflicts)
+    next._whitelist = this._whitelist
+      .union(schema._whitelist)
+      .setminus(schema._blacklist);
```

Comment:
this is doing a bit more work than needed, lets roll `union` and `setminus` into a single function, e.g. `merge(newItems, removeItems)`

Reply:
Done